src/librustc_index/vec.rs

```diff
@@ -182,12 +182,12 @@ macro_rules! newtype_index {
         impl Idx for $type {
             #[inline]
             fn new(value: usize) -> Self {
-                Self::from(value)
+                Self::from_usize(value)
```

Comment:
This replaces a trait method call by an inherent method call, increasing robustness against trait method ambiguities.